src/main/java/org/isf/menu/gui/MainMenu.java

```diff
@@ -348,61 +358,71 @@ private void launchApp(String itemMenuCode) {
 		private static final long serialVersionUID = 4338749100837551874L;
 
 		private JButton[] button;
-		private MainMenu parentFrame;
+		private MainMenu parentFrame = null;
 
 		public MainPanel(MainMenu parentFrame) {
 			this.parentFrame = parentFrame;
 			int numItems = 0;
 
-			for (UserMenuItem u : myMenu) {
-				if (u.getMySubmenu().equals("main")) {
+			for (UserMenuItem u : myMenu)
+				if (u.getMySubmenu().equals("main"))
 					numItems++;
-				}
-			}
+
+			// System.out.println(numItems);
+
 			button = new JButton[numItems];
 
-			int k = 0;
-			for (UserMenuItem u : myMenu) {
+			int k = 1;
+
+			for (UserMenuItem u : myMenu)
 				if (u.getMySubmenu().equals("main")) {
-					button[k] = new JButton(u.getButtonLabel());
-					button[k].setMnemonic(KeyEvent.VK_A + (u.getShortcut() - 'A'));
+					button[k - 1] = new JButton(u.getButtonLabel());
+
+					button[k - 1].setMnemonic(KeyEvent.VK_A + (int) (u.getShortcut() - 'A'));
 
-					button[k].addActionListener(parentFrame);
-					button[k].setActionCommand(u.getCode());
+					button[k - 1].addActionListener(parentFrame);
+					button[k - 1].setActionCommand(u.getCode());
 					k++;
 				}
-			}
 
+			setButtonsSize(button);
+
+			// setBackground(java.awt.Color.WHITE);
 			JLabel fig = new JLabel(new ImageIcon("rsc" + File.separator + "images" + File.separator + "LogoMenu.jpg"));
 			add(fig, BorderLayout.WEST);
 
 			JPanel buttons = new JPanel();
-			buttons.setLayout(new SpringLayout());
-			for (JButton jButton : button) {
-				buttons.add(jButton);
-			}
-			SpringUtilities.makeCompactGrid(buttons, button.length, 1, 6, 6, 8, 10);
-			add(buttons, BorderLayout.EAST);
-		}
-	}
+			GridBagLayout layout = new GridBagLayout();
+			buttons.setLayout(layout);
+
+			final int insetsValue = 6;
```

Comment:
the new standard on develop uses SprintUtilities

Reply:
the version in develop is the new look-n-feel

Reply:
I think that is an old code. I haven't dealt yet with the GUI editing.